### PR TITLE
[WFLY-17653] Upgrade OpenTelemetry to 1.23.1

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-17653

Bump version
Update archives to match upstream packageing reorgs

### CODE REVIEW

File: ee-feature-pack/common/src/main/resources/license/ee-feature-pack-common-licenses.xml [hunk 1]
```diff
@@ -826,6 +826,16 @@
         </license>
       </licenses>
     </dependency>
+    <dependency>
```
Comment (jamezp) [inline ee-feature-pack/common/src/main/resources/license/ee-feature-pack-common-licenses.xml:829]: Think we're missing the `opentelemetry-api-logs` dependency in here.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml [hunk 2]
```diff
@@ -25,6 +25,12 @@
     </properties>
     <resources>
         <artifact name="${io.opentelemetry:opentelemetry-api}"/>
+        <artifact name="${io.opentelemetry:opentelemetry-api-logs}"/>
+        <artifact name="${io.opentelemetry:opentelemetry-sdk}"/>
+        <artifact name="${io.opentelemetry:opentelemetry-sdk-logs}"/>
+        <artifact name="${io.opentelemetry:opentelemetry-sdk-metrics}"/>
+        <artifact name="${io.opentelemetry:opentelemetry-sdk-trace}"/>
+        <artifact name="${io.opentelemetry:opentelemetry-sdk-common}"/>
```
Comment (jamezp) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:33]: Is there a reason these were mostly moved out of `io.opentelemtry.trace`? This is explicitly named with "api" in the module name and these look like implementation details. Am I wrong?

**Review by jamezp — COMMENTED** [review]
~~Is~~ Not sure how this got there.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml (hunk 2 above)
Comment (jasondlee) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:33]: I moved those to make them visible to.. something -- I can't recall off hand; I think it's in the Micrometer PR. To be honest, these should probably be in their own module, though I'm trying to reduce the churn in this PR. If you think it's important enough, I can reorg those modules.

File: ee-feature-pack/common/src/main/resources/license/ee-feature-pack-common-licenses.xml (hunk 1 above)
Comment (jasondlee) [reply ee-feature-pack/common/src/main/resources/license/ee-feature-pack-common-licenses.xml:829]: Added.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml (hunk 2 above)
Comment (jamezp) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:33]: It feels a little weird to put the i the API IMO. I would opt more for a `io.opentelemetry.sdk` module over moving them here. Other opinions are welcome though :)

**Review by jamezp — APPROVED** [review]
I think this looks okay. It's a pretty big jump, but it sounds like we need to do it. 

@bstansberry the only minor question I'd have is if it's okay to rename `io.opentelemtry.trace` to `io.opentelemety.export`. It's a private module so it should be fine, but I wasn't sure if we have anything outside of WildFly using it. Maybe @jmesnil has an idea there as it would likely be something cloud related.

**Review by rhusar — CHANGES_REQUESTED** [review]
@jasondlee BTW the CI failures are related and need fixing:

    java.lang.Exception: Some unreferenced modules are un-expected [io.opentelemetry.sdk]

**Comment by jasondlee** [comment]
> @jasondlee BTW the CI failures are related and need fixing:
> 
> ```
> java.lang.Exception: Some unreferenced modules are un-expected [io.opentelemetry.sdk]
> ```

Right. Thanks. That's on my list for first thing Monday morning. :)

**Comment by bstansberry** [comment]
Thanks @jasondlee, @jamezp and @rhusar

**Pull request merged**